core/src/program/mod.rs

```diff
@@ -126,15 +127,26 @@ impl CodeBlockTable {
 #[derive(Debug, Clone, Default, PartialEq, Eq)]
 pub struct Kernel(Vec<Digest>);
 
+pub const MAX_KERNEL_PROCEDURES: usize = u8::MAX as usize;
+
 impl Kernel {
     /// Returns a new [Kernel] instantiated with the specified procedure hashes.
-    pub fn new(proc_hashes: &[Digest]) -> Self {
-        // make sure procedure roots are ordered consistently
-        let mut hash_map: BTreeMap<[u8; 32], Digest> = BTreeMap::new();
-        proc_hashes.iter().cloned().for_each(|r| {
-            hash_map.insert(r.into(), r);
-        });
-        Self(hash_map.values().copied().collect())
+    pub fn new(proc_hashes: &[Digest]) -> Result<Self, errors::KernelError> {
+        if proc_hashes.len() > MAX_KERNEL_PROCEDURES {
+            Err(errors::KernelError::TooManyProcedures(MAX_KERNEL_PROCEDURES, proc_hashes.len()))
+        } else {
+            let mut hashes = proc_hashes.to_vec();
+            hashes.sort(); // ensure consistent order
```

Comment:
I think this might be sorting using the Montgomery representation of field elements. I wonder if a better approach would be to sort by binary value of hashes (i.e., by `[u8; 32]`).

Reply:
changed